iothub_client/CMakeLists.txt

```diff
@@ -73,25 +56,38 @@ if (${use_prov_client})
     )
 endif()
 
-if(NOT ${dont_use_uploadtoblob})
-    set(iothub_client_ll_transport_h_files
-        ${iothub_client_ll_transport_h_files}
+if(NOT dont_use_uploadtoblob)
```

Comment:
>dont_use_uploadtoblob [](start = 7, length = 21)

Interesting that you removed the ${...} here.  Are you sure it works in all case?

Reply:
I believe that is actually how you are supposed to do it. I think, if by chance your variable contains the name of another variable, then that is the one it will test. Sometimes I've had cmake flag these as warnings.